airo-camera-toolkit/airo_camera_toolkit/point_clouds/operations.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any
+
+import numpy as np
+from airo_typing import BoundingBox3DType, PointCloud
+
+
+def filter_point_cloud(point_cloud: PointCloud, mask: Any) -> PointCloud:
+    """Creates a new point cloud that is filtered by the given mask.
+    Will also filter the colors and attributes if they are present.
+
+    Args:
+        point_cloud: the point cloud to filter
+        mask: the mask to filter the point cloud by, used to index the attribute arrays, can be boolean or indices
+
+    Returns:
+        the new filtered point cloud
+    """
+    points = point_cloud.points[mask]
+    colors = None if point_cloud.colors is None else point_cloud.colors[mask]
+
+    attributes = None
+    if point_cloud.attributes is not None:
+        attributes = {}
+        for key, value in point_cloud.attributes.items():
+            attributes[key] = value[mask]
+
+    point_cloud_filtered = PointCloud(points, colors, attributes)
+    return point_cloud_filtered
+
+
+def crop_point_cloud_mask(point_cloud: PointCloud, bounding_box: BoundingBox3DType) -> np.ndarray:
```

Comment:
would name this 'generate_point_cloud_mask_from_box' to highlight that it does not filter the pc yet

Reply:
Great idea!